tinyagentos/agent_chat_router.py

```
@@ -63,23 +112,34 @@ async def _route_inner(self, message: dict, channel: dict) -> None:
-            # POST /api/openclaw/sessions/{agent}/reply (handled in
-            # routes/openclaw.py), which writes traces and broadcasts to the
-            # chat hub -- no polling, no HTTP callback from this path.
+            context = []
+            if hasattr(self._state, "chat_messages"):
+                try:
+                    from tinyagentos.chat.context_window import build_context_window
+                    recent = await self._state.chat_messages.get_messages(
+                        channel_id=channel["id"], limit=30,
+                    )
+                    context = build_context_window(recent, limit=20, max_tokens=4000)
+                except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)